### PR TITLE
ENH expand boost matrix, not replace

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge/core This Pr would start the boost 1.72 migration over with expanding the build matrix. We could in principle keep the old migration number, but I worry a lot of deps have moved on in their versions and so they would not be available for 1.70 now because the 1.72 migration has been going so long.

File: recipe/migrations/boost172.yaml [hunk 1]
```diff
@@ -3,11 +3,12 @@ __migrator:
   kind:
     version
   migration_number:
-    2
+    3
```
Comment (isuruf) [inline recipe/migrations/boost172.yaml:6]: Changing this would make conda-smithy remove 1.72 until the new migration is hit right?

Comment (beckermr) [reply recipe/migrations/boost172.yaml:6]: The new migration would startup almost immediately and 172 builds already exist, so the initial ones should go fast. However, yes, AFAIK, you are correct.

Comment (isuruf) [reply recipe/migrations/boost172.yaml:6]: Wouldn't it be better to make a new 1.70_and_1.72 migrator and stop the 1.72 migrator manually without removing the yaml?

Comment (beckermr) [reply recipe/migrations/boost172.yaml:6]: Doing that is equivalent to change the pin on latest in this PR and some of the data in this file, right?

Comment (isuruf) [reply recipe/migrations/boost172.yaml:6]: What do you mean by changing some of the data in this file?

Comment (beckermr) [reply recipe/migrations/boost172.yaml:6]: Ohhhh just resetting the timestamp and renaming the file etc.

Comment (isuruf) [reply recipe/migrations/boost172.yaml:6]: No, if we do that, the feedstocks that have already been migrated will revert to 1.70 until the bot sends a PR with the new migration.

Comment (beckermr) [reply recipe/migrations/boost172.yaml:6]: We change the value in conda_build_config.yaml to 1.72 as well. 

This is what I mean by "change the pin on latest". 

Then feedstocks won't get reverted.

Comment (isuruf) [reply recipe/migrations/boost172.yaml:6]: Then feedstocks that haven't been migrated starts failing, right?

Comment (beckermr) [reply recipe/migrations/boost172.yaml:6]: What do you mean?

Comment (beckermr) [reply recipe/migrations/boost172.yaml:6]: We'd have the 1.70 migration going at the same time, backporting all of the pins we need.

Comment (isuruf) [reply recipe/migrations/boost172.yaml:6]: Feedstocks that haven't been migrated to 1.72 yet will start getting 1.72 before the bot has migrated their upstreams right?

Comment (beckermr) [reply recipe/migrations/boost172.yaml:6]: Let's just add another 1.70 and 1.72 migrator and try setting pause on the 1.72 one. Hold on.

Comment (beckermr) [reply recipe/migrations/boost172.yaml]: ok @isuruf - check out the latest version - the paused key stops the migration without removing currently migrated things.

**Comment by jakirkham** [comment]
cc @conda-forge/core (as we discussed this in the meeting today)

File: recipe/migrations/boost170_and_172.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+migrator_ts: 1595449217
+__migrator:
+  kind:
+    version
+  migration_number:
+    1
+  build_num2er:
```
Comment (jakirkham) [inline recipe/migrations/boost170_and_172.yaml:7]: Maybe I'm just missing something, but should this be?

```suggestion
  build_number:
```

**Comment by beckermr** [comment]
Closing for now.

**Pull request closed without merging**